internal/provider/vaultsecrets/vault_secrets_utils.go

```diff
@@ -58,22 +80,63 @@ var sharedIntegrationAttributes = map[string]schema.Attribute{
 	},
 }
 
+// managedSecretAttributes are attribute shared between secrets with a managed lifecycle, such as Dynamic or Rotating
+var managedSecretAttributes = map[string]schema.Attribute{
+	"app_name": schema.StringAttribute{
+		Description: "Vault Secrets application name that owns the secret.",
+		Required:    true,
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.RequiresReplace(),
+		},
+		Validators: []validator.String{
+			slugValidator,
+		},
+	},
+	"secret_provider": schema.StringAttribute{
+		Description: "The third party platform the dynamic credentials give access to. One of `aws` or `gcp`.",
+		Required:    true,
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.RequiresReplace(),
+		}, // TODO: Add validators check its in enum
```

Comment:
Still todo? I checked this case and received `unsupported provider, expected one of [gcp aws], got 'gcpp'`.

Reply:
Whoops, no its a leftover.